src/ui/connectiondialog.cc

```diff
@@ -173,7 +160,78 @@ void ConnectionDialog::serverFileSelected(const QString& file_name) {
   ui_->server_executable_line_edit->setText(file_name);
 }
 
+void ConnectionDialog::loadDefaultValues() {
+  ui_->new_server_radio_button->setChecked(
+      util::settings::connection::runServerDefault());
+  serverLocalhost();
+  ui_->port_spin_box->setValue(
+      util::settings::connection::serverPortDefault());
+  ui_->key_line_edit->setText("");
+  clientLocalhost();
+  ui_->client_name_line_edit->setText(
+          util::settings::connection::clientNameDefault());
+  ui_->database_line_edit->setText(
+      util::settings::connection::databaseNameDefault());
+  ui_->server_executable_line_edit->setText("");
+  ui_->shut_down_server_check_box->setChecked(
+      util::settings::connection::shutDownServerOnQuitDefault());
+}
+
+void ConnectionDialog::loadSettings() {
+  QSettings settings;
+
+  (util::settings::connection::runServer() ? ui_->new_server_radio_button
+      : ui_->existing_server_radio_button)->setChecked(true);
+  ui_->server_host_line_edit->setText(
+      util::settings::connection::serverHost());
+  ui_->port_spin_box->setValue(util::settings::connection::serverPort());
+  ui_->key_line_edit->setText(util::settings::connection::connectionKey());
+  ui_->client_interface_line_edit->setText(
+      util::settings::connection::clientInterface());
+  QString client_name = util::settings::connection::clientName();
+  ui_->client_name_line_edit->setText(client_name);
+  ui_->database_line_edit->setText(
+      util::settings::connection::databaseName());
+  ui_->server_executable_line_edit->setText(
+      util::settings::connection::serverScript());
+  ui_->shut_down_server_check_box->setChecked(
+      util::settings::connection::shutDownServerOnQuit());
+}
+
+void ConnectionDialog::saveSettings() {
+  QSettings settings;
+
+  util::settings::connection::setRunServer(
+      ui_->new_server_radio_button->isChecked());
+  util::settings::connection::setServerHost(
+      ui_->server_host_line_edit->text());
+  util::settings::connection::setServerPort(
+      ui_->port_spin_box->value());
+  util::settings::connection::setClientInterface(
+      ui_->client_interface_line_edit->text());
+  util::settings::connection::setClientName(
+      ui_->client_name_line_edit->text());
+  util::settings::connection::setConnectionKey(
+      ui_->save_key_check_box->isChecked() ?
+      ui_->key_line_edit->text() : "");
+  util::settings::connection::setDatabaseName(
+      ui_->database_line_edit->text());
+  util::settings::connection::setServerScript(
+      ui_->server_executable_line_edit->text());
+  util::settings::connection::setShutDownServerOnQuit(
+      ui_->shut_down_server_check_box->isChecked());
+}
+
+void ConnectionDialog::saveSettingsToggled(bool toggled) {
+  ui_->save_key_check_box->setEnabled(toggled);
+}
+
+void ConnectionDialog::dialogAccepted() {
+  saveSettings();
```

Comment:
Why doesn't `save_settings_check_box` state affect this? Right now save is unconditional.